File: DFS&BFS/javascript/11725.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');

const n = Number(input[0]);
let tree = Array.from(Array(n+1),() => new Array());
let parentNode = new Array(n+1).fill(0);
let a,b;

for (let i = 1 ; i<input.length-1 ; i++){
  [a,b] = input[i].split(' ').map(Number);
  tree[a].push(b);
  tree[b].push(a);
}


const bfs = ()=>{
  q = [];
  idx = 0;
  q.push(1);
  let node, nextNode;
  parentNode[0] = true;
  parentNode[1] = true;
  while (idx<q.length){
    node = q[idx];
    for (nextNode of tree[node]){
      if (parentNode[nextNode] !== 0) continue;
      parentNode[nextNode] = node;
      q.push(nextNode);
    }
    idx++;
  }
}
bfs()
let result = "";
for (let i = 2 ; i<=n ; i++){
  result+= parentNode[i];
  if (i !== n) result+="\n";
}
console.log(result);

File: DFS&BFS/python/11725.py
from collections import deque
import sys
input = sys.stdin.readline
n = int(input())
tree = [[] for _ in range(n+1)]
parent_node = [0 for _ in range(n+1)]

for _ in range(n-1):
  a,b = map(int,input().split())
  tree[a].append(b)
  tree[b].append(a)

def bfs(parent_node):
    q = deque()
    parent_node[1] = True
    parent_node[0] = True
    q.append(1)
    while q:
        node = q.popleft()
        for next_node in tree[node]:
            if parent_node[next_node] == 0 :
                parent_node[next_node] = node
                q.append(next_node)
    return parent_node
bfs(parent_node)
for i in range(2,n+1):
    print(parent_node[i])
